Sources/CADCacheAdvance/CADCacheAdvance.swift

```diff
@@ -98,30 +98,17 @@ public final class __ObjectiveCCompatibleCacheAdvanceWithGenericData: NSObject {
 /// A decoder that treats all messages as if they are `Data`.
 final class PassthroughDataDecoder: MessageDecoder {
     func decode<T>(_ type: T.Type, from data: Data) throws -> T where T : Decodable {
-        if let data = data as? T {
-            return data
-        } else {
-            throw DecodingError.dataCorrupted(
-                DecodingError.Context(
-                    codingPath: [],
-                    debugDescription: "Type was not Data"))
-        }
+        // Force cast because this type is only used with a CacheAdvance<Data> type.
```

Comment:
So it would be programmer error if someone used `PassthroughDataDecoder` with a result type other than `Data`?

Reply:
It would. That type is internal to the repo so there's not much chance of that happening.